app/controllers/api/providers_controller.rb

```diff
@@ -102,15 +102,17 @@ def change_password_resource(type, id, data = {})
       end
     end
 
-    # NOTE: _type == :providers
-    def verify_credentials_resource(_type, id = nil, data = {})
-      klass = fetch_provider_klass(collection_class(:providers), data)
-      zone_name = data.delete('zone_name')
-      data['id'] = id if id
-      task_id = klass.verify_credentials_task(current_user.userid, zone_name, data)
-      action_result(true, 'Credentials sent for verification', :task_id => task_id)
-    rescue => err
-      action_result(false, err.to_s)
+    # NOTE: the ui's zone drop down is currently passing the zone_id
+    def verify_credentials_resource(type, id = nil, data = {})
+      api_action(type, id) do |klass|
+        ems_klass = fetch_provider_klass(klass, data)
+        zone_name = data.delete('zone_name')
+        zone_id = data.delete('zone_id')
```

Comment:
Funny I was just looking into this when I saw my `zone` was blank for verifying credentials.  This seems like its own bug that we should fix separately so we can backport what do you think?

Reply:
see #1211